pkg/engine/engine.go

```diff
@@ -224,21 +246,29 @@ func (e *Engine) detectorWorker(ctx context.Context) {
 				if decoded == nil {
 					continue
 				}
+
 				dataLower := strings.ToLower(string(decoded.Data))
+				matches := e.prefilter.FindAll(dataLower)
```

Comment:
**Question:** if the `prefilter` is doing a case-insensitive match, do we still need to lowercase the data here?

Reply:
We could maybe use [strings.EqualFold](https://pkg.go.dev/strings#EqualFold) when comparing the matches against the keywords.

Reply:
@mcastorina I believe we do still need to lowercase the data here since we are using `matchedKeywords` as a lookup which expect all keywords to be lowercase. 

> We could maybe use [strings.EqualFold](https://pkg.go.dev/strings#EqualFold) when comparing the matches against the keywords.

I may be misunderstanding but using `strings.EqualFold` would require iterating over all the matched keywords which would be linear time vs a constant time map lookup. Did you have an implementation specifically in mind?

`strings.EqualFold` is nice though, I'm sure we could introduce this in other parts of the code! Found a good article on string comparisons in go: https://hackernoon.com/optimizing-string-comparisons-in-go-7h1b3udm

Reply:
Ah yeah, that makes sense. I didn't think it through fully, but you're right that `EqualFold` wouldn't work with a map lookup.

Perhaps only lower-casing the keys to the map would work though, instead of the entire chunk of data. That should function the same, but save on memory (and one less pass of the data).